script/validators/source_validator.py

```diff
@@ -202,6 +204,39 @@ def check_includes(file_path):
     return file_status
 
 
+def check_tests(file_path):
+    """Checks test source files for correct class and method naming."""
+    # check class naming
+    class_status = True  # For reporting class names.
+    test_status = True  # For reporting test cases.
+    line_num = 0
+    with open(file_path, "r") as file:
+        for line in file:
+            line_num += 1
+            if line.startswith('class '):
+                class_name = line.split(' ')[1]
+                if not class_name.endswith('Tests'):
+                    if class_status:
+                        LOG.info("Invalid class name in %s", file_path)
+                        class_status = False
+                    LOG.info("Line %s: %s", line_num, line.strip())
+
+            else:
```

Comment:
Should there be a separate loop for this? That will prevent interleaved output.

Reply:
I am ok with this, we hope not to have thousands of those errors.